### PR TITLE
Update diagrams for sovereign sequencer and confirmation layer

### PR DESCRIPTION
**Pull request by sveitser**
- Add the overview and sequence diagrams from our docs and update them.
- Add the integration diagrams from our docs and update them.
- Move architecture description to its own file.
- Rename "Espresso Sequencer" to "Espresso" resp. "Confirmation Layer".

The two simplified overview diagrams and the integration diagrams are currently visible at: https://docs.espressosys.com/sequencer/espresso-architecture/system-overview

### CODE REVIEW

File: doc/full-node-sequencer-integration.svg
Comment (jbearer) [inline doc/full-node-sequencer-integration.svg:1]: Don't know if it matters but should we remove `sequencer` from this file name?

File: doc/prover-sequencer-integration.svg
Comment (jbearer) [inline doc/prover-sequencer-integration.svg:1]: and here

**Review by jbearer — APPROVED** [review]

File: doc/full-node-sequencer-integration.svg
Comment (sveitser) [reply doc/full-node-sequencer-integration.svg:1]: might as well

Comment (sveitser) [reply doc/full-node-sequencer-integration.svg:1]: https://github.com/EspressoSystems/espresso-sequencer/pull/2220/commits/4977dfee26cb32b463a58e52536a44af5cf3785b

File: doc/prover-sequencer-integration.svg
Comment (sveitser) [reply doc/prover-sequencer-integration.svg:1]: https://github.com/EspressoSystems/espresso-sequencer/pull/2220/commits/4977dfee26cb32b463a58e52536a44af5cf3785b

**Pull request merged**